ocaml/xapi/xapi_host.ml

```diff
@@ -1737,8 +1737,12 @@ let sync_tunnels ~__context ~host =
           ()
         | [(pif_ref,_)] ->
           (* this is the PIF on which we want as transport PIF; let's make it *)
+          let (_, tunnel_rec) = List.hd (Db.Tunnel.get_records_where ~__context ~expr: (
```

Comment:
`List.hd` will raise an exception on an empty list, please treat that case with a match structure instead to raise an appropriate error:
```
let tunnel_rec = match Db.tunnel.get_ ... with
| (_, tunnel_rec) :: _ -> tunnel_rec
| [] -> raise Server_error ...
```

Reply:
Done! :)